solc/CommandLineInterface.cpp

```diff
@@ -692,7 +692,15 @@ void CommandLineInterface::compile()
 {
 	solAssert(CompilerInputModes.count(m_options.input.mode) == 1);
 
-	m_compiler = make_unique<CompilerStack>(m_fileReader.reader());
+	auto modelCheckerSettings = m_options.modelChecker.settings;
+	if (modelCheckerSettings.solvers.eld)
+	{
+		m_solverCommand = make_unique<SMTSolverCommand>("eld");
+		m_universalCallback = make_unique<UniversalCallback>(m_fileReader, *m_solverCommand);
+		m_compiler = make_unique<CompilerStack>(m_universalCallback->callback());
+	}
```

Comment:
I just realized that this will only run on the CLI. What about Standard JSON?

Also, since the new solver is in model checker settings, we'll actually accept `eld` in Standard JSON but just do nothing.

Reply:
Standard JSON is tricky because we can't be sure whether it's running from the command line or the browser, for example.

Reply:
Running via Standard JSON with `eld` we get:
```
"Warning: CHC analysis was not possible. No Horn solver was available. None of the installed solvers was enabled.\n\n"
```

Reply:
ok, we do not exactly do nothing because part of it is handled in `ModelChecker`. So the warning will still appear.

I checked with solc-js and I also get the message about `eld` not being available in the system, but I think that may be confusing to the user. It does not make it clear that it's not allowed, user is likely to think he's just doing it wrong and waste time tinkering with it. After all, with `--standard-json` on the CLI it technically could work so I think we should really have a validation against `eld` being selected as one of the solvers in Standard JSON.

And anyway, using filesystem stuff like `boost::process::search_path("eld")` anywhere outside of the CLI (e.g. in `ModelChecker.cpp`) is questionable in general. Can we even assume that emscripten will always handle that in a reasonable way? To be honest, I'm surprised that it does not raise some kind of exception or crash.

Reply:
Ah yea true... that's actually the reason why we moved it entirely into the CLI before I added that eld bit. Sigh this is so annoying.

Reply:
@cameel we can't easily do such validation and modification of the JSON input in the CLI because there it's still just a string. Here's an idea I talked about with Chris today:

- we don't do any of the CLI shenanigans, no internal flag in settings saying whether the `eld` command is available or not
- if the user selected `eld` as a solver, we assume it's available and create the proper `UniversalCallback`
- when we make the call to the cmd, we `ifdef` check whether we're running in emscripten mode and don't even try to make the call if yes
- otherwise we make the call, and if it turns out the binary is not available it fails properly, which is fine